### PR TITLE
Feature: 로그인 기능 구현 완료

### LINKED ISSUE #3: 💡 Feature: 로그인 기능 구현하기
## Key Feature

### PR DESCRIPTION
**Pull request by yeonise**
## Key Feature

로그인에 필요한 기능들을 구현했습니다.

1. 암호화 관련 기능

- `학번` : _AES256_ - 양방향 암호화하여 저장
- `비밀번호` : _BCrypt_ - 단방향 암호화하여 저장

  관련 클래스들은 `util` 패키지에 생성하였습니다.
- **Aes256Manager**
- **BCryptManager**

2. 세션 관련 기능

- **SessionManager** : `auth.application` 패키지에 생성하였습니다.
  - 세션 생성 및 저장
  - 세션 조회
  - 세션 갱신
  - 세션 무효화

- **SessionStorage** : `auth.infrastructure` 패키지에 생성하였습니다.
  - CrudRepository 상속
  - Redis 연결

3. 로그인 기능

- 학번과 비밀번호로 로그인을 요청합니다 → 인증에 성공하면 세션을 생성하고 Redis에 저장합니다 → 세션 아이디를 쿠키에 담아 응답합니다.

4. 쿠키 생성

- Controller에서 생성하여 `Set-Cookie` 헤더에 담아서 전송합니다.
- `HttpOnly(true)` : Sets the flag that controls if this cookie will be hidden from scripts on the client side.
- `Secure(true)` : Indicates to the browser whether the cookie should only be sent using a secure protocol, such as HTTPS or SSL.

5. Annotation을 사용한 DTO 검증
6. 기본적인 예외 처리 기능
7. REST Docs 문서 생성

(#3)

### CODE REVIEW

**Comment by ghkdgus29** [comment]
### Test Coverage Report
|Overall Project|92.93% **`-4.72%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|94.46%|:green_apple:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[RedisConfig.java](https://github.com/mujik-tigers/course-registration-system/blob/e8bc0023509586b570e18ed353f0b2c80e9cba95/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fconfig%2FRedisConfig.java)|100%|:green_apple:|
|[PropertiesConfig.java](https://github.com/mujik-tigers/course-registration-system/blob/e8bc0023509586b570e18ed353f0b2c80e9cba95/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fconfig%2FPropertiesConfig.java)|100%|:green_apple:|
|[AuthController.java](https://github.com/mujik-tigers/course-registration-system/blob/e8bc0023509586b570e18ed353f0b2c80e9cba95/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fauth%2Fpresentation%2FAuthController.java)|100%|:green_apple:|
|[Aes256Properties.java](https://github.com/mujik-tigers/course-registration-system/blob/e8bc0023509586b570e18ed353f0b2c80e9cba95/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Futil%2Fencryption%2FAes256Properties.java)|100%|:green_apple:|
|[CustomException.java](https://github.com/mujik-tigers/course-registration-system/blob/e8bc0023509586b570e18ed353f0b2c80e9cba95/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2FCustomException.java)|100%|:green_apple:|
|[GlobalExceptionHandler.java](https://github.com/mujik-tigers/course-registration-system/blob/e8bc0023509586b570e18ed353f0b2c80e9cba95/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2FGlobalExceptionHandler.java)|100%|:green_apple:|
|[ErrorType.java](https://github.com/mujik-tigers/course-registration-system/blob/e8bc0023509586b570e18ed353f0b2c80e9cba95/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2FErrorType.java)|100%|:green_apple:|
|[StudentSession.java](https://github.com/mujik-tigers/course-registration-system/blob/e8bc0023509586b570e18ed353f0b2c80e9cba95/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fauth%2FStudentSession.java)|100%|:green_apple:|
|[ResponseMessage.java](https://github.com/mujik-tigers/course-registration-system/blob/e8bc0023509586b570e18ed353f0b2c80e9cba95/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Futil%2Fapi%2FResponseMessage.java)|100%|:green_apple:|
|[PasswordFormatCheck.java](https://github.com/mujik-tigers/course-registration-system/blob/e8bc0023509586b570e18ed353f0b2c80e9cba95/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fvalidator%2FPasswordFormatCheck.java)|100%|:green_apple:|
|[StudentIdFormatCheck.java](https://github.com/mujik-tigers/course-registration-system/blob/e8bc0023509586b570e18ed353f0b2c80e9cba95/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fvalidator%2FStudentIdFormatCheck.java)|100%|:green_apple:|
|[InvalidPasswordException.java](https://github.com/mujik-tigers/course-registration-system/blob/e8bc0023509586b570e18ed353f0b2c80e9cba95/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2Fauth%2FInvalidPasswordException.java)|100%|:green_apple:|
|[NonexistenceStudentIdException.java](https://github.com/mujik-tigers/course-registration-system/blob/e8bc0023509586b570e18ed353f0b2c80e9cba95/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2Fauth%2FNonexistenceStudentIdException.java)|100%|:green_apple:|
|[NonexistenceSessionException.java](https://github.com/mujik-tigers/course-registration-system/blob/e8bc0023509586b570e18ed353f0b2c80e9cba95/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fexception%2Fauth%2FNonexistenceSessionException.java)|100%|:green_apple:|
|[SessionManager.java](https://github.com/mujik-tigers/course-registration-system/blob/e8bc0023509586b570e18ed353f0b2c80e9cba95/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fauth%2Fapplication%2FSessionManager.java)|100%|:green_apple:|
|[AuthService.java](https://github.com/mujik-tigers/course-registration-system/blob/e8bc0023509586b570e18ed353f0b2c80e9cba95/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fauth%2Fapplication%2FAuthService.java)|100%|:green_apple:|
|[LoginForm.java](https://github.com/mujik-tigers/course-registration-system/blob/e8bc0023509586b570e18ed353f0b2c80e9cba95/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fauth%2Fdto%2FLoginForm.java)|100%|:green_apple:|
|[Aes256Manager.java](https://github.com/mujik-tigers/course-registration-system/blob/e8bc0023509586b570e18ed353f0b2c80e9cba95/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Futil%2Fencryption%2FAes256Manager.java)|85.19% **`-14.81%`**|:green_apple:|
|[ApiResponse.java](https://github.com/mujik-tigers/course-registration-system/blob/e8bc0023509586b570e18ed353f0b2c80e9cba95/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Futil%2Fapi%2FApiResponse.java)|83.72% **`-16.28%`**|:green_apple:|
|[BCryptManager.java](https://github.com/mujik-tigers/course-registration-system/blob/e8bc0023509586b570e18ed353f0b2c80e9cba95/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Futil%2Fencryption%2FBCryptManager.java)|72.73% **`-27.27%`**|:green_apple:|
|[Student.java](https://github.com/mujik-tigers/course-registration-system/blob/e8bc0023509586b570e18ed353f0b2c80e9cba95/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fstudent%2FStudent.java)|60%|:green_apple:|
|[CookieProperties.java](https://github.com/mujik-tigers/course-registration-system/blob/e8bc0023509586b570e18ed353f0b2c80e9cba95/src%2Fmain%2Fjava%2Fsite%2Fcourseregistrationsystem%2Fauth%2Fpresentation%2FCookieProperties.java)|55.56% **`-44.44%`**|:green_apple:|

File: src/test/java/site/courseregistrationsystem/auth/application/SessionManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import site.courseregistrationsystem.exception.ErrorType;
+import site.courseregistrationsystem.exception.auth.NonexistenceSessionException;
+
+class SessionManagerTest extends IntegrationTestSupport {
+
+	@Autowired
+	private SessionManager sessionManager;
+
+	@Test
+	void generate() {
```
Comment (ghkdgus29) [inline src/test/java/site/courseregistrationsystem/auth/application/SessionManagerTest.java:19]: @Display 는 귀찮아서 안달았나요?!

Comment (ghkdgus29) [reply src/test/java/site/courseregistrationsystem/auth/application/SessionManagerTest.java:19]: 봐드리겠습니다

**Comment by ghkdgus29** [comment]
수고하셨습니다!!
레디스를 crud Repository 를 사용해 깔끔하게 추상화한 모습이 인상적이네요!

File: src/test/java/site/courseregistrationsystem/auth/application/SessionManagerTest.java (hunk 1 above)
Comment (yeonise) [reply src/test/java/site/courseregistrationsystem/auth/application/SessionManagerTest.java:19]: ㅋㅋㅋ마지막에 한번에 한다는걸 까먹었습다 얌 🍊
다음 PR에 함께 작성하여 올리겠습니다

Comment (yeonise) [reply src/test/java/site/courseregistrationsystem/auth/application/SessionManagerTest.java:19]: **@** 사용 시 백틱 처리를 해줘야 멘션이 되지 않습니다 (부끄)

**Pull request merged**